src/Raven.Client/Documents/Replication/ReplicationNode.cs

```diff
@@ -13,7 +13,7 @@ namespace Raven.Client.Documents.Replication
     /// <summary>
     /// Data class for replication destination
     /// </summary>
-    public abstract class ReplicationNode : IEquatable<ReplicationNode>
```

Comment:
Why `abstract` got dropped?

Reply:
I see in the error that `Can't compile a NewExpression with a constructor declared on an abstract class` so I supposed it's because of that. But my question is then why it attempted to create an instance of this class?

Reply:
from my understanding, we generate a `NewExpression` for the `ReplicationNode` type, which allows us to convert the `blittableJsonObject` response to the corresponding type using its constructor. I've spent a lot of time attempting to find a workaround for this issue, but it seems that this approach is necessary in this context. 
the only potential solution I've found is to explicitly define the underlying class (for instance, `InternalReplication` instead of `ReplicationNode`). but, this approach isn't going to work here since we don't have knowledge of the specific type at this stage, and there are multiple options to consider.

Reply:
What is the exact line of  code causing the issue?

Reply:
I think https://github.com/ravendb/ravendb/blob/v6.0/src/Sparrow/Json/JsonDeserializationBase.cs#L45
then we get the failure from https://github.com/ravendb/ravendb/blob/v6.0/src/Sparrow/Json/JsonDeserializationBase.cs#L127

Reply:
Something that I cannot understand is why / where do we create the deserialization routine for `ReplicationNode` type (which is an abstract class)?

Reply:
The creation of deserialization routines for types like `ReplicationNode` happens during the initialization phase. Initially, we generate JSON deserialization routines for each object defined in `JsonDeserializationServer.cs` and its nested objects, recursively. for instance, https://github.com/ravendb/ravendb/blob/v6.0/src/Raven.Server/Json/JsonDeserializationServer.cs#L101, where one of the parameters is `ReplicationNode`.

when a call is made, such as from `RavenCommand.SetResponse`, for example `Result = JsonDeserializationServer.ReplicationActiveConnectionsPreview(response);`, we retrieve the compiled function to execute for this specific type. 
initially, before the fix, we encountered `FailureBuildingJsonParser` during the execution of `GenerateJsonDeserializationRoutine`, which ultimately led to an exception during deserialization.

Reply:
I have doubts if dropping `abstract` is the solution. As you pointed the deserialization routines are built during the initialization phase so they are build based on properties / fields of a given type. As I understand the issue with `ReplicationActiveConnectionsPreview` is that is has `public List<ReplicationNode> OutgoingConnections;` and `ReplicationNode` is an abstract class:

https://github.com/ravendb/ravendb/blob/0ddbe3215a3e6d4814758f9bb4211d3cb9a6f93b/src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractReplicationHandlerProcessorForGetActiveConnections.cs#L25-L29

But dropping `abstract` I understand that we no longer get the exception during the build of the deserializaition routine (`GenerateJsonDeserializationRoutine<ReplicationActiveConnectionsPreview>()`). But the routing will be built only for the filelds of `ReplicationNode` type, wont' it? What about additional fields of types which inherit from `ReplicationNode`?

What I think _might_ be a solution is `IFillFromBlittableJson` interface. Please check it and ping me if you need some assistance with that.